lib/index.js

```diff
@@ -39,7 +39,13 @@ internals.implementation = function (server, options) {
       // otherwise use the same key (String) to validate all JWTs
       var decoded;
       try {
-        decoded = JWT.decode(token); // decode is non-io and fast enough to not have to be async
+		if (options.completeToken === true) { 
```

Comment:
For backward compatibility with current users of this plugin, I really wanted that if `completeToken` isn't exactly defined as `true`, the returned value of `decoded` is still defined as `JWT.decode(token)`, without any option passed at all (even not a false `complete` argument) in this underlying call.
For the same reason, the default should be that only the payload is returned (i.e. true shoudn't become the default).

In terms of performance, when the token is decoded in plugin (node-jws)[https://github.com/brianloveswords/node-jws], all 3 parts are always decoded, so so impact shouldn't be noticeable at all.